internal/payloads/error_payload.go

```diff
@@ -27,30 +32,40 @@ func (e *ErrResponse) Error() error {
 	return fmt.Errorf("%s (%d): %w", e.Message, e.Code, e.Err)
 }
 
-func ErrInvalidRequest(err error) render.Renderer {
+func ErrRender(err error, ctx context.Context) render.Renderer {
 	return &ErrResponse{
 		Err:            err,
-		HTTPStatusCode: 400,
-		Message:        "Invalid request",
+		HTTPStatusCode: 422,
+		RequestId:      ctx.Value(ctxval.RequestIdCtxKey).(string),
+		Message:        "Unable to render response",
 	}
 }
 
-func ErrAWSGeneric(err error) render.Renderer {
+func ErrAcquireDAO(err error, ctx context.Context) render.Renderer {
+	return &ErrResponse{
+		Err:            err,
+		HTTPStatusCode: 500,
+		RequestId:      ctx.Value(ctxval.RequestIdCtxKey).(string),
+		Message:        "Unable to acquire DAO implementation",
+	}
+}
+
+func ErrParamParsing(param string, err error, ctx context.Context) render.Renderer {
 	return &ErrResponse{
 		Err:            err,
 		HTTPStatusCode: 400,
-		Message:        "Error returned from AWS",
+		RequestId:      ctx.Value(ctxval.RequestIdCtxKey).(string),
+		Message:        fmt.Sprintf("Invalid request: unable to parse parameter: %s", param),
 	}
 }
 
-func ErrRender(err error) render.Renderer {
+func ErrFetchingRecord(err error, ctx context.Context) render.Renderer {
 	return &ErrResponse{
 		Err:            err,
-		HTTPStatusCode: 422,
-		Message:        "Error rendering response.",
+		RequestId:      ctx.Value(ctxval.RequestIdCtxKey).(string),
+		HTTPStatusCode: 500,
+		Message:        "Cannot fetch record from DB",
 	}
 }
 
 var ErrNotFound = &ErrResponse{HTTPStatusCode: 404, Message: "Resource not found"}
-var ErrParamParsingError = &ErrResponse{HTTPStatusCode: 500, Message: "Cannot parse parameters"}
-var ErrDeleteError = &ErrResponse{HTTPStatusCode: 500, Message: "Cannot delete resource"}
```

Comment:
Yeah errors package needs more love, I am on that in my other PR.

Reply:
https://github.com/RHEnVision/provisioning-backend/pull/6/files#diff-4b50e8af15cd6a0dc76bc19957e33530d7bc3d431b0c1c16f1ad022dbfdca2a7